core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -70,32 +71,20 @@
   public static final String DEFAULT_PROPERTIES = "alluxio-default.properties";
   /** File to set customized properties. */
   public static final String SITE_PROPERTIES = "alluxio-site.properties";
-  /** Regex string to find ${key} for variable substitution. */
+  /** Regex string to find "${key}" for variable substitution. */
   private static final String REGEX_STRING = "(\\$\\{([^{}]*)\\})";
   /** Regex to find ${key} for variable substitution. */
   private static final Pattern CONF_REGEX = Pattern.compile(REGEX_STRING);
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
-
+  /** Set of properties */
   private final Properties mProperties = new Properties();
 
   /**
    * Overrides default properties.
    *
    * @param props override {@link Properties}
    */
-  public Configuration(Map<String, String> props) {
-    if (props != null) {
-      mProperties.putAll(props);
-    }
-    checkUserFileBufferBytes();
-  }
-
-  /**
-   * Overrides default properties.
-   *
-   * @param props override {@link Properties}
-   */
-  public Configuration(Properties props) {
+  public Configuration(Map<?, ?> props) {
```

Comment:
same as above, this should be the goal and let's see how far we go in this PR (not having a too big PR)